src/agoradatatools/gx.py

```diff
@@ -16,13 +19,19 @@ class GreatExpectationsRunner:
     """Class to run great expectations on a dataset and upload the HTML report to Synapse"""
 
     def __init__(
-        self, syn: Synapse, dataset_path: str, dataset_name: str, upload_folder: str
+        self,
+        syn: Synapse,
+        dataset_path: str,
+        dataset_name: str,
+        upload_folder: str,
+        nested_columns: list = [],
```

Comment:
Let's avoid having empty lists as a default value.   I think may need to use the defaultfactory feature.

Reply:
I have access. Just hadn't gotten back around to the issues it flagged yet. I'll make this change.

Reply:
I made this change, which resolved the `Critical` issue that sonarcloud raised. The other suggestions are either nonsensical (return types for unit tests), or unable to be added due to Python 3.8 support (subscripted type objects)